### PR TITLE
feat(AppHeaderDropdown): dropdown AppHeaderElement

### PR DESCRIPTION
**Pull request by katiezutter**
### Overview

Create AppHeaderDropdown component to be part of AppHeaderElements.

image

Part of the [Global Nav Unification](https://codecademy.atlassian.net/browse/EGG-642) project.

This PR comes after [EGG-539](https://codecademy.atlassian.net/browse/EGG-539) and [EGG-540](https://codecademy.atlassian.net/browse/EGG-540), work done with @saghdaey which resulted in a [draft PR combining all GlobalHeader work](https://github.com/Codecademy/client-modules/pull/1265). We are now breaking up this branch to merge the following pieces separately:

1. [EGG-643](https://codecademy.atlassian.net/browse/EGG-643): AppHeaderLogo (#1277)
2. [EGG-644](https://codecademy.atlassian.net/browse/EGG-644): AppHeaderLink (#1278)
3. [EGG-645](https://codecademy.atlassian.net/browse/EGG-645): AppHeaderDropdown (#1279)
4. [EGG-646](https://codecademy.atlassian.net/browse/EGG-646): AppHeader
5. [EGG-647](https://codecademy.atlassian.net/browse/EGG-647): GlobalHeader

This PR adds unit tests to AppHeaderDropdown.

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [EGG-645](https://codecademy.atlassian.net/browse/EGG-645)
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/__tests__/AppHeaderDropdown-test.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+describe('AppHeaderDropdown', () => {
+  it('displays target link text', () => {
+    renderAppHeaderDropdown();
+    expect(screen.getByText(testTargetText));
+  });
+
+  it('target links to the provided href', () => {
+    renderAppHeaderDropdown();
+  });
```
Comment (saghdaey) [inline packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/__tests__/AppHeaderDropdown-test.tsx:51]: think you may have forgotten to add the test checking the provided href :)

Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/__tests__/AppHeaderDropdown-test.tsx:51]: lol 🤦‍♀️

Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/__tests__/AppHeaderDropdown-test.tsx:51]: I think I just copied that over from the link tests - I think that behavior of opening the dropdown by clicking the target is tested by Popover! So I'll just remove it

Comment (saghdaey) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/__tests__/AppHeaderDropdown-test.tsx]: sounds good to me!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.3b03cb.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60006a89d65f4f00b5f01f84--gamut-preview.netlify.app>https://60006a89d65f4f00b5f01f84--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60006a89d65f4f00b5f01f84>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.660f91.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.d53fe4.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6000a80af968983ec5183266--gamut-preview.netlify.app>https://6000a80af968983ec5183266--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6000a80af968983ec5183266>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        isOpen={isOpen}
+        onRequestClose={handleClose}
+        targetRef={headerDropdownRef}
+      >
+        <Box paddingY={12}>
+          {item.popover.map((link) => {
+            return (
+              <Box key={link.id} paddingX={16}>
+                <AppHeaderLink
+                  // className={styles.menuItem} - className isn't being used in AppHeaderLinkElement so commenting this out
```
Comment (hmafzal) [inline packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:93]: Can we remove this?

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  &:hover {
+    color: ${theme.colors.hyper};
+    text-decoration: none;
+  }
+  ${focusStyles}
+`;
+});
+
+const StyledPopover = styled(Popover)`
+  z-index: 100;
```
Comment (hmafzal) [inline packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:36]: Why is this necessary? Seems a little off to have a z-index defined in this component, since we don't know what the max z-index could be in the consuming application. Maybe this should be passed in by consumer through props?

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+`;
+});
+
+const AppHeaderLinkButtonInner = styled(Box)`
+  white-space: nowrap;
+`;
+
+export type AppHeaderLinkProps = {
+  item: AppHeaderLinkItem;
+  // onClick: HeaderClickHandler;
```
Comment (hmafzal) [inline packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderLink/index.tsx:27]: Similar comments from other PR for these

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx (hunk 3 above)
Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:36]: Good point! I would say we could remove - @saghdaey any objections there?

File: packages/gamut-labs/src/experimental/AppHeader/types.tsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+type AppHeaderBaseItem = {
+  dataTestId?: string;
+  id: string;
+  text?: string;
+};
+
+export type AppHeaderLinkItem = AppHeaderBaseItem & {
+  href: string;
+  trackingTarget: string;
```
Comment (hmafzal) [inline packages/gamut-labs/src/experimental/AppHeader/types.tsx:9]: The trackingTarget prop feels a little weird to me. Does that mean that the call to the track() function will happen within the gamut component? I would think that it should happen through a callback onClick prop that the consumer passes.

I see that onClick will be a part of the props in the future, so why is trackingTarget necessary?

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx (hunk 3 above)
Comment (saghdaey) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:36]: the content under the popover was showing for me when the dropdown was rendered, adding z index fixed that

Comment (saghdaey) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:36]: maybe we just do z index 1? popover didnt have a z index prop

File: packages/gamut-labs/src/experimental/AppHeader/types.tsx (hunk 5 above)
Comment (saghdaey) [reply packages/gamut-labs/src/experimental/AppHeader/types.tsx:9]: the track function will be called by the clickHandler function defined in the monolith

Comment (saghdaey) [reply packages/gamut-labs/src/experimental/AppHeader/types.tsx:9]: https://github.com/Codecademy/client-modules/pull/1265/files see HeaderClickHandler
we provide "item" to the header click handler, that's why we need the trackingTarget

Comment (saghdaey) [reply packages/gamut-labs/src/experimental/AppHeader/types.tsx:9]: monolith passes HeaderClickHandler to global header --> 

export type HeaderClickHandler = (
  event: React.MouseEvent,
  item: AppHeaderItem
) => void;

monolith isn't rendering the app header items - its just providing 2 props : a variant (determines which header to load) + the onClick function that all the links will use

export type GlobalHeaderProps = {
  variant: AnonHeader | FreeHeader | ProHeader;
  onClick: HeaderClickHandler;
};

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.7.5-alpha.053bd3.0<br/>@codecademy/styleguide@21.10.1-alpha.053bd3.0<br/></pre>

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx (hunk 3 above)
Comment (hmafzal) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:36]: Even defaulting a z-index of 1 assumes things about what zindexs are of elements on the page this is used in the consuming app.

I think you could have z-index be a prop of AppHeaderDropdownProps if it's absolutely necessary

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6000b434fe07e458917e0ca3--gamut-preview.netlify.app>https://6000b434fe07e458917e0ca3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6000b434fe07e458917e0ca3>Deploy Logs</a></p>

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+export const AppHeaderDropdown: React.FC<AppHeaderDropdownProps> = ({
+  item,
+  onClick,
+}) => {
+  const headerDropdownRef = useRef<HTMLDivElement>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const toggleIsOpen = (event: React.MouseEvent) => {
+    setIsOpen(!isOpen);
+    !isOpen && onClick(event); // TODO pass item through
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:52]: Is there some context for this TODO?

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6000c2d314eaf300e7ab8026--gamut-preview.netlify.app>https://6000c2d314eaf300e7ab8026--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6000c2d314eaf300e7ab8026>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.8.1-alpha.7eb912.0<br/>@codecademy/styleguide@21.11.1-alpha.7eb912.0<br/></pre>

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx (hunk 6 above)
Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:52]: Yes! It relates to the TODO above ([here](https://github.com/Codecademy/client-modules/pull/1279/files#diff-da6e62ca12094f8efb8c2d8d4302f7c252d2732769a564ee94f1603f059b3c6bR41))

We thought it'd be easier to do it this way until we create the PR for the GlobalHeader component, which has the `HeaderClickHandler` type: https://github.com/Codecademy/client-modules/pull/1265/files#diff-5dc98ac2d2a55c876938b81d81d874ff90110bbe94ce3c992aad650090112847R13

I can alternatively add this type somewhere now if that makes more sense!

Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:52]: Either works for me :) no strong preference

File: packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderDropdown.stories.mdx [hunk 7]
```diff
@@ -0,0 +1,42 @@
+import { Meta, Props, Canvas, Story } from '@storybook/addon-docs/dist/blocks';
+import { AppHeaderDropdown } from '@codecademy/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx';
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderDropdown.stories.mdx:2]: Oh, just noticing these now:

```suggestion
import { AppHeaderDropdown } from '@codecademy/gamut-labs/src';
```

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+import React, { useRef, useState } from 'react';
+
+import { Popover } from '../../..';
+import { AppHeaderLink } from '../AppHeaderLink';
+import { focusStyles } from '../SharedStyles';
+import { AppHeaderDropdownItem, AppHeaderLinkItem } from '../types';
+import styles from './styles.scss';
+
+const AppHeaderTargetButton = styled.button(({ theme }) => {
+  return `
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:14]: This should have `cursor: pointer`. It's unusual for clickable elements to not, and `button`s don't have it by default for some bizarre reason.

Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:14]: ...TIL: https://ux.stackexchange.com/questions/105024/why-dont-button-html-elements-have-a-css-cursor-pointer-by-default#:~:text=pointer%20The%20cursor%20is%20a,have%20to%20do%20it%20manually.

**Review by JoshuaKGoldberg — APPROVED** [review]
Hasan is already asking all of the questions I would have. This is great overall!

File: packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx (hunk 3 above)
Comment (katiezutter) [reply packages/gamut-labs/src/experimental/AppHeader/AppHeaderElements/AppHeaderDropdown/index.tsx:36]: Taking this out for now - made a note to revisit later when we put everything together!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.8.1-alpha.d7384d.0<br/>@codecademy/styleguide@21.11.1-alpha.d7384d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6000ce84301e100f5306e778--gamut-preview.netlify.app>https://6000ce84301e100f5306e778--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6000ce84301e100f5306e778>Deploy Logs</a></p>

File: packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderDropdown.stories.mdx (hunk 7 above)
Comment (saghdaey) [reply packages/styleguide/stories/Labs/Experimental/Molecules/AppHeaderDropdown.stories.mdx]: good catch

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.8.1-alpha.fff001.0<br/>@codecademy/styleguide@21.11.1-alpha.fff001.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001103b04ca0a6859fbd3b3--gamut-preview.netlify.app>https://6001103b04ca0a6859fbd3b3--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001103b04ca0a6859fbd3b3>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://600117910c48c36fe202da45--gamut-preview.netlify.app>https://600117910c48c36fe202da45--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/600117910c48c36fe202da45>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.8.1-alpha.e0be81.0<br/>@codecademy/styleguide@21.11.1-alpha.e0be81.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-labs@7.9.1-alpha.5f5f1f.0<br/>@codecademy/styleguide@21.12.1-alpha.5f5f1f.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6001b40b5fd95429902e05d1--gamut-preview.netlify.app>https://6001b40b5fd95429902e05d1--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6001b40b5fd95429902e05d1>Deploy Logs</a></p>
